### PR TITLE
Bump python 3.10 -> 3.11 and nixpkgs

### PR DESCRIPTION
**Pull request by bryango**
Nixpkgs has been using python 3.11 for some time already, so bumping the python version used here to catch up! This would save a few hundred megabytes when hydra-check is installed in the system. Also, take the chance to bump everything else:

- bump nixpkgs, use python 3.11
- poetry2nix now out of tree, import as flake input
- add minor cleanup of flake.nix
- update poetry dependencies

I have checked basic functionalities and it seems to work well!

### CODE REVIEW

**Comment by bryango** [comment]
Can we also do a point release (i.e. `1.3.5 -> 1.3.6`) such that I can bump `nixpkgs.hydra-check` accordingly? :pleading_face: Thank you!

**Comment by bryango** [comment]
Friendly ping @makefu and @Artturin for review :wink:

File: flake.nix [hunk 1]
```diff
@@ -3,52 +3,57 @@
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
-    flake-utils = { url = "github:numtide/flake-utils"; };
-    flake-compat = {
```
Comment (Artturin) [inline flake.nix:7]: You dropped flake-compat

Comment (bryango) [reply flake.nix:7]: Oh no! Sorry... Thank you for noticing this!

**Pull request merged**